pkg/controller/types/networkpolicy.go

```diff
@@ -38,9 +38,9 @@ type GroupSelector struct {
 	Namespace string
 	// This is a label selector which selects pods. If Namespace is also set, it selects the pods in the namespace.
 	// If NamespaceSelector is also set, it selects the pods in the namespaces selected by NamespaceSelector.
-	PodSelector *metav1.LabelSelector
+	PodSelector *labels.Selector
```

Comment:
labels.Selector is an interface so can be assigned nil, no need to use pointer for unset case, then the above code can be simpler:
```
pSelector := sel.PodSelector
if !(*pSelector).Matches(labels.Set(pod.Labels)) {
```
to 
```
if !sel.PodSelector.Matches(labels.Set(pod.Labels)) {
```